File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Masters/getmasterinfo.rb
module Asposediagramjava
  module GetMasterInfo
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Call the diagram constructor to load diagram from a VSD file
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        masters = diagram.getMasters()

        i = 0
        while i < masters.getCount()
            master = masters.get(i)
            puts "Master ID : " + master.getID().to_s
            puts "Master Name : " + master.getName().to_s

            i +=1
        end        
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithMasters/GetMasterInfo.php
<?php
namespace Aspose\Diagram\WorkingwithMasters;

use com\aspose\diagram\Diagram as Diagram;
class GetMasterInfo{

    public static function run($dataDir=null)
    {

        # Call the diagram constructor to load diagram from a VSD file
        $diagram = new Diagram($dataDir . "drawing.vsd");

        $masters = $diagram->getMasters();

        $i = 0;
        while ($i<sizeof($masters->getCount())){
            $master = $masters->get($i);
            print "Master ID : " . (string)$master->getID().PHP_EOL;
            print "Master Name : " . (string)$master->getName().PHP_EOL;
            $i += 1;
        }
    }
}